### PR TITLE
rename self.exe attribute

### PR DESCRIPTION
**Pull request by bblay**
Actioning [this review comment](https://github.com/metomi/fab/pull/145#discussion_r979864912), self.exe is renamed for better readability.

### CODE REVIEW

**Review by MatthewHambley — APPROVED** [review]
A simple change which improves readability.

File: source/fab/steps/compile_fortran.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def __init__(self, compiler: str = None, common_flags: List[str] = None, path_fl
         compiler = compiler or os.getenv('FC', 'gfortran -c')
         common_flags = common_flags or []
         env_flags = os.getenv('FFLAGS', '').split()
-        self.exe: str = compiler or os.getenv('FC', 'gfortran -c')  # type: ignore
+        self.compiler: str = compiler or os.getenv('FC', 'gfortran -c')  # type: ignore
```
Comment (MatthewHambley) [inline source/fab/steps/compile_fortran.py:68]: We seem to have two instances of `compiler or os.getenv('FC', 'gfortran -c')`, here and line 65. Is that intended?

**Review by MatthewHambley — CHANGES_REQUESTED** [review]
Noticed an oddity

File: source/fab/steps/compile_fortran.py (hunk 1 above)
Comment (bblay) [reply source/fab/steps/compile_fortran.py:68]: It's my use of the magic wand merge button again. I _did_ say I would be more vigilant with it's use...sorry!

**Review by MatthewHambley — APPROVED** [review]
Doubled up compiler determination fixed.

**Review by MatthewHambley — APPROVED** [review]
As noted by the developers, many of the changes originally in this patch have been swept up in others that there isn't much left.

**Pull request merged**